server/game/abilities/keyword/AmbushAbility.ts

```diff
@@ -1,36 +1,54 @@
 import AbilityHelper from '../../AbilityHelper';
+import { AbilityContext } from '../../core/ability/AbilityContext';
 import TriggeredAbility from '../../core/ability/TriggeredAbility';
+import { TriggeredAbilityContext } from '../../core/ability/TriggeredAbilityContext';
 import { Card } from '../../core/card/Card';
+import { UnitCard } from '../../core/card/CardTypes';
 import { KeywordName } from '../../core/Constants';
 import Game from '../../core/Game';
 import * as Contract from '../../core/utils/Contract';
 import { ITriggeredAbilityProps } from '../../Interfaces';
 
-/** @deprecated this is still WIP until attack changes are done */
-// export class AmbushAbility extends TriggeredAbility {
-//     public override readonly keyword: KeywordName | null = KeywordName.Ambush;
+export class AmbushAbility extends TriggeredAbility {
+    public override readonly keyword: KeywordName | null = KeywordName.Ambush;
 
-//     public static buildAmbushAbilityProperties(): ITriggeredAbilityProps {
-//         return {
-//             title: 'Ambush',
-//             optional: true,
-//             when: { onUnitEntersPlay: (event, context) => event.card === context.source },
-//             immediateEffect: [
-//                 AbilityHelper.immediateEffects.ready(),
-//                 AbilityHelper.immediateEffects.attack((context) => {
-//                     return { attacker: context.source };
-//                 })
-//             ]
-//         };
-//     }
+    public static buildAmbushAbilityProperties<TSource extends Card = Card>(): ITriggeredAbilityProps<TSource> {
+        return {
+            title: 'Ambush',
+            optional: true,
+            when: { onUnitEntersPlay: (event, context) => event.card === context.source },
+            immediateEffect: AbilityHelper.immediateEffects.conditional({
+                condition: AmbushAbility.unitWouldHaveAmbushTarget<TSource>,
+                onTrue: AbilityHelper.immediateEffects.sequential([
+                    AbilityHelper.immediateEffects.ready(),
+                    AbilityHelper.immediateEffects.attack((context) => {
```

Comment:
Is there a way this attack could be tagged as being due to ambush? Would be useful for set 2 Enfys Nest.

Reply:
I think that's a good idea but for a simple feature like that I like waiting until that card that needs it is being implemented so we can write and test it in the same PR. I'll add a TODO here in the meantime